components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -864,6 +864,64 @@ void BraveWalletService::MigrateUserAssetsAddIsNFT(PrefService* prefs) {
   prefs->SetBoolean(kBraveWalletUserAssetsAddIsNFTMigrated, true);
 }
 
+// static
+void BraveWalletService::MigrateUserAssetsAddFilecoinEvm(PrefService* prefs) {
+  if (prefs->GetBoolean(kBraveWalletFilecoinEvmNetworksAdded)) {
+    return;
+  }
+
+  {
+    const std::string filecoin_evm_mainnet =
+        GetKnownEthNetworkId(mojom::kFilecoinEtheriumMainnetChainId);
+    const std::string filecoin_evm_testnet =
+        GetKnownEthNetworkId(mojom::kFilecoinEtheriumTestnetChainId);
+
+    ScopedDictPrefUpdate update(prefs, kBraveWalletUserAssets);
+    auto& user_assets_pref = update.Get();
+    auto* dict = user_assets_pref.EnsureDict(kEthereumPrefKey);
+
+    // Filecoin mainnet
+    {
+      base::Value::List* list = dict->EnsureList(filecoin_evm_mainnet);
+      auto value = base::Value(GetEthNativeAssetFromChain(
+          GetKnownChain(nullptr, filecoin_evm_mainnet, mojom::CoinType::ETH)));
+      if (std::find_if(list->begin(), list->end(), [&value](auto& v) {
+            return value == v;
+          }) == list->end()) {
+        list->Insert(list->begin(), std::move(value));
+      }
+    }
+
+    // Filecoin testnet
+    {
+      base::Value::List* list = dict->EnsureList(filecoin_evm_testnet);
+      auto value = base::Value(GetEthNativeAssetFromChain(
+          GetKnownChain(nullptr, filecoin_evm_testnet, mojom::CoinType::ETH)));
+      if (std::find_if(list->begin(), list->end(), [&value](auto& v) {
+            return value == v;
+          }) == list->end()) {
+        list->Insert(list->begin(), std::move(value));
+      }
+    }
+  }
+
+  // Default hidden networks
+  {
+    ScopedDictPrefUpdate update(prefs, kBraveWalletHiddenNetworks);
+    auto& hidden_networks_pref = update.Get();
+    base::Value::List* hidden_eth_networks =
+        hidden_networks_pref.FindList(kEthereumPrefKey);
+    auto value = base::Value(mojom::kFilecoinEtheriumTestnetChainId);
+    if (std::find_if(hidden_eth_networks->begin(), hidden_eth_networks->end(),
+                     [&value](auto& v) { return value == v; }) ==
+        hidden_eth_networks->end()) {
+      hidden_eth_networks->Append(std::move(value));
+    }
+  }
```

Comment:
Could we make this part generalized and have a function named similar as `MigrateHiddenNetworksAddPreloadingTestnets`, and re-used it every time we're adding a new test network?